### PR TITLE
update value as needed and reuse variables

### PR DESCRIPTION
**Pull request by minggo**
Re: https://github.com/cocos/cocos-engine/issues/16952

### Changelog

* reuse local variable
* use static method of  Vec3.set() and Quat.set() as they don't have to distinguish variable types, so they are more efficient
* make Node.setWorldPosition(), Node.setWorldRotation(), Node.setWorldRotationFromEuler(), Node.setWorldScale() update values add needed

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by minggo** [comment]
@cocos-robot run test cases

File: cocos/scene-graph/node.ts [hunk 1]
```diff
@@ -2482,19 +2521,30 @@ export class Node extends CCObject implements ISchedulable, CustomSerializable {
...
-            Vec3.copy(this._scale, val as Vec3);
+        if (y === undefined) {
+            if (worldScale.equals(val as Vec3)) {
+                return;
+            }
+
+            Vec3.copy(worldScale, val as Vec3);
         } else {
-            Vec3.set(this._scale, val as number, y, z);
+            Vec3.set(v3_a, val as number, y, z!);
```
Comment (dumganhar) [inline cocos/scene-graph/node.ts:2537]: What happens if val is number, y is number, z is undefined?

Comment (minggo) [reply cocos/scene-graph/node.ts:2537]: It doesn't meet the function declaration.

Comment (dumganhar) [reply cocos/scene-graph/node.ts:2537]: Got it. I forgot it was an overload function

**Review by dumganhar — APPROVED** [review]

**Comment by minggo** [comment]
@cocos-robot run test cases

**Pull request merged**